plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,275 @@
+       point pairs that straddle :math:`I = 0`.  This forms an array of
+       "crossing-points."
+    #. The crossing-points are then grouped into "crossing-islands" in based on
+       the ``threshold`` keyword.
+
+       - A new island is formed when a successive crossing-point is more (index)
+         steps away from the previous crossing-point than allowed by
+         ``threshold``.
+       - If multiple crossing-islands are identified, then the total span of all
+         crossing-islands is compared to ``min_points``.  If the span is greater
+         than ``min_points`` then the function is incapable of identifying
+         :math:`V_f` and will return `numpy.nan` values; otherwise, the span
+         will form one larger crossing-island.
+
+    #. To calculate the floating potential...
+
+       - If the crossing-island contains less points than ``min_points``, then
+         each side of the crossing-island is equally padded with the nearest
+         neighbor points until ``min_points`` is satisfied.
+       - A fit is then performed using `scipy.stats.linregress` for
+         ``fit_type="linear"`` and `scipy.optimize.curve_fit` for
+         ``fit_type="exponential"``.
+
+    Parameters
+    ----------
+
+    voltage: ~numpy.ndarray
+        1-D numpy array of monotonically ascending/descending probe biases
+        (in Volts)
+
+    current: ~numpy.ndarray
+        1-D numpy array of probe current (in A) corresponding to the
+        :data:`voltage` array
+
+    threshold: positive, non-zero `int`
+        Max allowed index distance between crossing-points before a new
+        crossing-island is formed.  That is, if `threshold=5` then consecutive
+        crossing-points are considered to be in the same crossing-island if
+        they are within 5 index steps of each other. (Default: 1)
+
+    min_points: positive `int` or `float`
+        Specifies the minimum number of points required for the fit to be
+        applied to.
+
+        - ``min_points = None`` (Default) The larger of 5 and
+          ``factor * array_size`` is taken, where ``array_size`` is the size of
+          ``voltage`` and ``factor = 0.1`` for ``fit_type = "linear"`` and
+          ``0.2`` for ``"exponential"``.
+        - ``min_points = 0`` The entire passed array is fitted.
+        - ``min_points >= 1`` Exact minimum number of points.
+        - ``0 < min_points < 0`` The minimum number of points is taken as
+          ``min_points * array_size``.
+
+    fit_type: str
+        The type of curve to be fitted to (``voltage``, ``current``).  There
+        are two types of curves ``"linear"`` and ``"exponential"`` (Default).
+
+    Returns
+    -------
+    vf: `float` or `numpy.nan`
+        The calculated floating potential (in Volts).  Returns `numpy.nan` if the
+        floating potential can not be determined.
+
+    vf_err: `float` or `numpy.nan`
+        The error associated with the floating potential calculation (in Volts).
```

Comment:
```suggestion
        The calculated floating potential (in volts).  Returns `numpy.nan` if the
        floating potential can not be determined.

    vf_err: `float` or `numpy.nan`
        The error associated with the floating potential calculation (in volts).
```

Reply:
I looked this up a while back and SI units are lower case even when they're based off of a name, except for Celsius.

Reply:
I don't know why I did it like that.   Probably just doing "in V...oh, that's not descriptive enough, Let's complete that...in Volts".